addons/binding/org.openhab.binding.nuki/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="nuki"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+        xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+    <!-- Nuki Bridge (Bridge Type) -->
+    <bridge-type id="bridge">
+        <label>Nuki Bridge</label>
+        <description>This bridge represents a Nuki Bridge on your local network. Nuki Smart Locks have to be paired via Bluetooth with it.</description>
+        <config-description>
+            <parameter name="ip" type="text" required="true">
+                <label>IP Address</label>
+                <description>The IP Address of the Nuki Bridge.</description>
+            </parameter>
+            <parameter name="port" type="integer" required="true">
+                <label>Port</label>
+                <description>The Port on which the Nuki Bridge REST API runs.</description>
+                <default>8080</default>
+            </parameter>
+            <parameter name="apiToken" type="text" required="true">
+                <label>API Token</label>
+                <description>The API Token configured via the Nuki App when enabling the API/IFTTT.</description>
+            </parameter>
+        </config-description>
+    </bridge-type>
+
+    <!-- Nuki Smart Lock (Thing Type) -->
+    <thing-type id="smartLock">
+        <supported-bridge-type-refs>
+            <bridge-type-ref id="bridge"/>
+        </supported-bridge-type-refs>
+        <label>Nuki Smart Lock</label>
+        <description>Nuki Smart Lock which is paired via Bluetooth to a Nuki Bridge.</description>
+        <channels>
+            <channel id="smartLockOpenClose" typeId="channelTypeSmartLockOpenClose"/>
+            <channel  id="smartLockUnlatchClose" typeId="channelTypeSmartLockUnlatchClose"/>
+        </channels>
+        <config-description>
+            <parameter name="nukiId" type="text" required="true">
+                <label>Nuki ID</label>
+                <description>The ID of the Nuki Smart Lock.</description>
+            </parameter>
+        </config-description>
+    </thing-type>
+
+    <!-- Channel Type -->
+    <channel-type id="channelTypeSmartLockOpenClose">
+        <item-type>Switch</item-type>
+        <label>Nuki Smart Lock Open/Close</label>
+        <description>Nuki Smart Lock channel to open or close the door.</description>
+    </channel-type>
+
+    <channel-type id="channelTypeSmartLockUnlatchClose">
+        <item-type>Switch</item-type>
+        <label>Nuki Smart Lock Unlatch/Close</label>
+        <description>Nuki Smart Lock channel to unlatch or close the door.</description>
```

Comment:
Well, you would like to dig deeper, ok :)
Actually the Nuki Smart Lock allows 2 "open" actions: "unlock" and "unlatch"
Furthermore it allows 2 "open and then close again" actions:  "lock 'n' go" and "lock 'n' go with unlatch"
And finally it allows 1 "lock" action.
Please see Bridge HTTP-API (https://nuki.io/en/api/), Section "Lock actions"

I am planning to implement:

1. Two channels of type Switch to cover the most basic usage which is obviously opening (unlock or unlatch depending on the door lock) and locking a door.
So one who has a knob uses just the channel "unlatch".
But one who has a handle could use both channels (even on the same sitemap): "unlock" to just unlock the door and then manually pulling the handle to open the door. And also "unlatch" if he wants to unlock and open the door via a Switch.
With either one or both of these channels one can just start using the Nuki Smart Lock without any technical knowledge (e.g. rules).

2. Two channels of type Switch to cover "lock 'n' go" (same reason as for point 1)

3. One channel (e.g. "lockAction") of type Number which accepts all possible "Lock actions" for technical advanced users. One could do something like this https://community.openhab.org/t/new-openhab2-binding-for-nuki-smart-locks/25940/30?u=mkatter

4. One channel (e.g. "lockState") of type String to display the current Lock state (see Bridge HTTP-API, Section Lock states)

5. A system.low-battery channel

6. A system.signal-strength


> functional difference between "close" of channel 1 and 2?
No, "close" is in any case Lock action 2

> Is UNLATCH(ED) a valid state or merely a command?
Please see Bridge HTTP-API, Section Lock states for vaild states. Please note: The Bridge at the moment is not returning all of these states in its current implementation but that might change in the future. So it's a good idea to implement the channel "lockState" right away.